spec/01-unit/000-new-dao/01-schema/01-schema_spec.lua

```diff
@@ -1276,10 +1277,10 @@ describe("schema", function()
       }))
       Test.primary_key = { "f" }
       local ok, errs = Test:validate_primary_key({
-        f = { id = ngx.null },
+        f = { },
```

Comment:
More of a question than a comment: I failed to understand how the changes in this commit affected the behavior of this test requiring this change. Could you elaborate a bit on this here, just so we're on the same page?

Reply:
You were right to raise questions about this!

The test were failing so I changed them to something "equivalent" without thinking.

It turns out there was actually a problem: the `_cache` variable I created was a bit "overzealous" in the way it "protected the cache": if you created a schema named `A` and later a second schema named the same way, *the cache would not replace the old schema with the new one*. Since the tests's reuse the same name again and again, the foreign keys were pointing to schemas from previous tests.

I've changed the cache so that it is less "protectionist" and now the tests work as before. Thanks!